src/org/jitsi/impl/neomedia/transform/fec/FlexFec03Receiver.java

```diff
@@ -0,0 +1,372 @@
+         * this is a reference so it will remain up to date as the map is
+         * filled out by the caller.
+         */
+        Reconstructor(Map<Integer, RawPacket> mediaPackets)
+        {
+            this.mediaPackets = mediaPackets;
+        }
+
+        public boolean complete()
+        {
+            return numMissing == 0;
+        }
+
+        public boolean canRecover()
+        {
+            return numMissing == 1;
+        }
+
+        public void setFecPacket(FlexFec03Packet p)
+        {
+            logger.debug("Have " + mediaPackets.size() + " saved media packets");
+            numMissing = 0;
+            logger.debug("Reconstructor checking if recovery is " +
+                "possible: fec packet " + p.getSequenceNumber() +
+                " protects packets:\n" + p.getProtectedSequenceNumbers());
+            this.fecPacket = p;
+            if (p == null)
+            {
+                logger.error("Error creating flexfec packet");
+                return;
+            }
+
+            for (Integer protectedSeqNum : fecPacket.getProtectedSequenceNumbers())
+            {
+                logger.debug("Checking if we've received media " +
+                    "packet " + protectedSeqNum);
+                if (!mediaPackets.containsKey(protectedSeqNum))
+                {
+                    logger.debug("We haven't, mark as missing");
+                    numMissing++;
+                    missingSequenceNumber = protectedSeqNum;
+                }
+            }
+            logger.debug("There were " + numMissing + " missing media " +
+                "packets for flexfec " + p.getSequenceNumber());
+            if (numMissing > 1)
+            {
+                missingSequenceNumber = -1;
+            }
+        }
+
+        /**
+         * Initialize the given RawPacket with the RTP header information
+         * and payload from fecPacket
+         * @param fecPacket the FlexFEC packet being used for recovery
+         * @param recoveredPacket the blank RawPacket we're recreating the
+         * recovered packet in
+         * @return true on success, false otherwise
+         */
+        private boolean startPacketRecovery(FlexFec03Packet fecPacket, RawPacket recoveredPacket)
+        {
+            if (fecPacket.getLength() < FlexFec03Packet.FIXED_HEADER_SIZE)
+            {
+                logger.error("Given FlexFEC packet is too small");
+                return false;
+            }
+            if (recoveredPacket.getBuffer().length < RawPacket.FIXED_HEADER_SIZE)
+            {
+                logger.error("Given RawPacket buffer is too small");
+                return false;
+            }
+            // Copy over the recovery RTP header data from the fec packet
+            // (fecPacket contains the RTP header, so we need to copy from it
+            // starting after that)
+            System.arraycopy(fecPacket.getBuffer(), fecPacket.getHeaderLength(),
```

Comment:
I think this needs to take the fecPacket offset into account

Reply:
fixed